integration/networktest/network_test_types.go

```diff
@@ -0,0 +1,56 @@
+package networktest
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// NetworkConnector represents the network being tested against, e.g. testnet, dev-testnet, dev-sim
+//
+// It provides network details (standard contract addresses) and easy client setup for sim users
+type NetworkConnector interface {
+	ChainID() int64
+	// AllocateFaucetFunds uses the networks default faucet mechanism for allocating funds to a test account
+	AllocateFaucetFunds(account common.Address) error
+	SequencerRPCAddress() string
+	ValidatorRPCAddress(idx int) string
+	NumValidators() int
+}
+
+// NetworkTest defines a test that can be run against an arbitrary network connector (same tests can run against different environments)
+type NetworkTest interface {
+	Run(t *testing.T, network NetworkConnector)
+}
+
+// DevNetwork interface represents an obscuro network that can be manipulated by tests
+// (currently there is just an in-mem implementation but one day it would be nice to be able to orchestrate the same
+// tests against SGX-enabled VMs)
+type DevNetwork interface {
+	NetworkConnector
+	Start()
+	GetSequencerNode() NodeOperator
+	GetValidatorNode(idx int) NodeOperator
+	CleanUp()
+}
+
+// ScenarioTest defines a test that can be run against a DevNetwork, e.g. to explore failover scenarios, restarts etc.
+type ScenarioTest interface {
```

Comment:
I'm not convinced about these ScenarioTest and NetworkTest abstractions.

The naming is definitely not right. 
"NetworkTest" - means you don't control nodes
"ScenarioTest" - means you control nodes

The point of these interfaces is to be able to create some batteries of tests that run in a sequence?